app/components/Poll/index.js

```diff
@@ -98,15 +120,14 @@ class Poll extends React.Component<Props, State> {
           <Flex column className={styles.optionWrapper}>
             <table className={styles.pollTable}>
               <tbody>
-                {optionsToShow.map(option => {
-                  const ratio = (options.length > 2 ? Math.floor : Math.round)(
-                    (option.votes / totalVotes) * 100
-                  );
+                {this.perfectRatios(this.optionsWithRatios(options, optionsToShow)).map(({id, name, votes, ratio}) => {
```

Comment:
I would like this line to be:
this.optionsWithRatios(options, optionsToShow).perfectRatios().map(…

but haven't been able to figure out how this is done in Javascript (or if it is possible).

Reply:
Thats not possible as far as i know, but I think it would be better to just have one function which calculates the perfect ratio, since the float version only is used for calculating the ratios that are shown.
So just have `this.perfectRatios(options, optionsToShow).map(...)` and just add the ratio property in the one function.